charts/fluent-bit/templates/servicemonitor.yaml

```diff
@@ -21,6 +22,14 @@ spec:
       {{- with .Values.serviceMonitor.scrapeTimeout }}
       scrapeTimeout: {{ . }}
       {{- end }}
+{{- if .Values.serviceMonitor.metricRelabelings }}
+      metricRelabelings:
+{{ tpl (toYaml .Values.serviceMonitor.metricRelabelings | indent 6) . }}
```

Comment:
I think this should follow the conditional template pattern where `tpl` is only called if the value is a string otherwise it's `toyaml`.

Reply:
fixed as suggested

Reply:
Do you not want to conditionally template this if the user provides a string?

An [example](https://github.com/stevehipwell/helm-charts/blob/5ddee3271d367f2631e1142c3cd6bb4a7b1acbab/charts/tigera-operator/templates/installation.yaml#L9) of the pattern.

Reply:
You probably want to use a with here too. It should be indented to 8 and there should be a reference in the values.

Reply:
I think now I did it as you suggested.

Also let me say that I follow the fluentd [servicemonitor.yaml](https://github.com/fluent/helm-charts/blob/2d2d83ec141c687313e8ddf1df80055606bc2f8e/charts/fluentd/templates/servicemonitor.yaml#L25)

In any case let me say thank you for your  comments.. I forgot it before....